Add section to FAQ to describe how `generate=False` may speed up RGBs (#2448)
pytroll/satpy: doc/source/faq.rst

    :depth: 1
    :local:

> How can I speed up image production for composites that need resampling?

Reviewer: ```suggestion
How can I speed up creation of composites that need resampling?
```

I'd like to avoid language that assumes we're generate "images". I don't want to confuse people who aren't generating images (PNG, JPEG) as their final output that this is only an option when writing to disk. Some people may also not consider creating interactive plots as generating images. Also, this doesn't only apply to RGB/A composites but any composite and any modifier.

------------------------------------------------------------------------

Satpy performs some initial image generation on the fly, but for composites
that need resampling (like the ``true_color`` composite for GOES/ABI) the data
must be resampled to a common grid before the final image can be producted, as
the input channels are at differing spatial resolutions. In such cases, you may
see a substantial performance improvement by passing `generate=False` when you
load your composite:

.. code-block:: python

    scn = Scene(filenames=filenames, reader='abi_l1b')
    scn.load(['true_color'], generate=False)
    scn_res = scn.resample(...)

> By default, `generate=True` which means that Satpy will compute some initial
> information about the scene at *each* spatial resolution.
> By setting `generate=False`, Satpy will only compute this information when the
> final image is generated. This can save a lot of time, and can result in images
> being generated up to 3x faster.

Reviewer: ```suggestion
By default, `generate=True` which means that Satpy will create as many
composites as it can with the available data. In some cases this could
a lot of intermediate products (ex. rayleigh corrected data using dynamically
generated angles for each band resolution) that will then need to be
resampled.
By setting `generate=False`, Satpy will only load the necessary dependencies
from the reader, but not attempt generating any composites or applying any
modifiers. In these cases this can save a lot of time and memory as only one
resolution of the input data have to be processed. Note that this option has
no effect when only loading data directly from readers (ex. IR/visible bands
directly from the files) and where no composites or modifiers are used. Also
note that in cases where most of your composite
inputs are already at the same resolution and you are only generating a limited
number of composites, ``generate=False`` may actually hurt performance.
```

    :depth: 1
    :local:

How can I speed up creation of composites that need resampling?
------------------------------------------------------------------------

Satpy performs some initial image generation on the fly, but for composites
that need resampling (like the ``true_color`` composite for GOES/ABI) the data
must be resampled to a common grid before the final image can be producted, as
the input channels are at differing spatial resolutions. In such cases, you may
see a substantial performance improvement by passing `generate=False` when you
load your composite:

.. code-block:: python

    scn = Scene(filenames=filenames, reader='abi_l1b')
    scn.load(['true_color'], generate=False)
    scn_res = scn.resample(...)

By default, ``generate=True`` which means that Satpy will create as many

> composites as it can with the available data. In some cases this could

Reviewer: ```suggestion
composites as it can with the available data. In some cases this could mean
```

Reviewer: Hhhmm or maybe `this could create`?
